### PR TITLE
Habbening - RG35XX-H Performance Optimisations and Button Mappings

### PR DESCRIPTION
**Pull request by Habbening**
# New Port for Jedi Knight : Jedi Academy

## Game Information
- **Title**: Jedi Knight : Jedi Academy
- **URL**: https://github.com/Cebion/PortMaster-New/tree/main/ports/jediacademy

### CFW Tests
Ensure your game has been tested on all major CFWs:
- [ ] AmberELEC
- [ ] ArkOS
- [ ] ROCKNIX
- [X] muOS

### Resolution Tests
Test all major resolutions:
- [ ] 480x320
- [X] 640x480
- [ ] 720x720 (RGB30)
- [ ] Higher resolutions (e.g., 1280x720)

## File Structure
- Your port should have the following structure:
  - portname/
    - port.json
    - README.md
    - screenshot.jpg
    - cover.jpg
    - gameinfo.xml
    - Port Name.sh
    - portname/
      - <portfiles here>

## Additional Resources
For an in-depth guide on creating a pull request, refer to: [PortMaster Game Packaging Guide](https://portmaster.games/packaging.html#creating-a-pull-request)

### CODE REVIEW

File: ports/jediacademy/Star Wars Jedi Knight - Jedi Academy.sh [hunk 1]
```diff
@@ -1,3 +1,4 @@
+#testing purposes - Habbening
```
Comment (Cebion) [inline ports/jediacademy/Star Wars Jedi Knight - Jedi Academy.sh:1]: Needs deleting =)

File: ports/jediacademy/JediAcademy/openjk_sp.aarch64.gptk [hunk 2]
```diff
@@ -1,28 +1,28 @@
...
-left_analog_down = s
-left_analog_left = a
-left_analog_right = d
-right_analog_up = mouse_movement_up
-right_analog_down = mouse_movement_down
-right_analog_left = mouse_movement_left
-right_analog_right = mouse_movement_right
+back =  n                                       # Quick Load
+start = m                                       # Quick Save
+guide = esc                                     # Pause / Menu
```
Comment (Cebion) [inline ports/jediacademy/JediAcademy/openjk_sp.aarch64.gptk:3]: On most devices there is no guide button. Guide is often synonymous with back. What button did you use on your rg35xxh to test this?

File: ports/jediacademy/JediAcademy/conf/openjk/base/openjk_sp.cfg [hunk 3]
```diff
@@ -92,36 +94,34 @@ seta g_saberLockRandomNess "2"
...
-seta g_spskill "0"
+seta g_spskill "1"
 seta g_stepSlideFix "1"
 seta g_subtitles "1"
 seta in_mouse "1"
 seta m_pitch "0.022000"
 seta r_centerWindow "0"
-seta r_customheight "1152"
-seta r_customwidth "1920"
+seta r_customheight "480"
```
Comment (Cebion) [inline ports/jediacademy/JediAcademy/conf/openjk/base/openjk_sp.cfg:106]: Does this break on devices other than 640x480?
I think I'm overriding these anyway in the .sh with :

```
sed -i "s/seta r_customheight \".*\"/seta r_customheight \"$DISPLAY_HEIGHT\"/" "$GAMEDIR/conf/openjo/base/openjo_sp.cfg"
sed -i "s/seta r_customwidth \".*\"/seta r_customwidth \"$DISPLAY_WIDTH\"/" "$GAMEDIR/conf/openjo/base/openjo_sp.cfg"```

**Review by Cebion — CHANGES_REQUESTED** [review]

File: ports/jediacademy/JediAcademy/conf/openjk/base/openjk_sp.cfg (hunk 3 above)
Comment (Habbening) [reply ports/jediacademy/JediAcademy/conf/openjk/base/openjk_sp.cfg:106]: Your correct, I also don't know why that still contains the original 1920x1152. I'll verify the whole file.

File: ports/jediacademy/JediAcademy/openjk_sp.aarch64.gptk (hunk 2 above)
Comment (Habbening) [reply ports/jediacademy/JediAcademy/openjk_sp.aarch64.gptk:3]: Guide = Function
Back = Select

File: ports/jediacademy/Star Wars Jedi Knight - Jedi Academy.sh (hunk 1 above)
Comment (Habbening) [reply ports/jediacademy/Star Wars Jedi Knight - Jedi Academy.sh:1]: you DO read them huh :P

**Review by Habbening — COMMENTED** [review]
Guide = Function
Back = Select
Start = Start

File: ports/jediacademy/Star Wars Jedi Knight - Jedi Academy.sh (hunk 1 above)
Comment (Habbening) [reply ports/jediacademy/Star Wars Jedi Knight - Jedi Academy.sh]: addressed

File: ports/jediacademy/JediAcademy/conf/openjk/base/openjk_sp.cfg (hunk 3 above)
Comment (Habbening) [reply ports/jediacademy/JediAcademy/conf/openjk/base/openjk_sp.cfg:106]: updated

**Review by Cebion — APPROVED** [review]

**Comment by Habbening** [comment]
@Cebion should I implement https://github.com/PortsMaster/Device-Info/blob/main/device_info.txt to ensure keybinds are for https://github.com/PortsMaster/Device-Info/blob/main/device_info.txt before proceeding?

**Comment by Cebion** [comment]
> @Cebion should I implement https://github.com/PortsMaster/Device-Info/blob/main/device_info.txt to ensure keybinds are for RG35xx-H before proceeding?

The RG35xx-H uses the same keylayout as ther other +20 devices.
The only difference is that many devices don't have a guide button.
Hell on some devices the FN key is just R3 if no analog sticks are present.

**Pull request merged**